### DISCUSSION TITLE
how to update

### DISCUSSION THREAD

**Discussion by 7by24** [madMAx43v3r/chia-plotter#687] [Q&A]
hi guys, I don't find any document of update procedure but only clean_all.sh. Do I need to run it and download the newest code then install it again?

**Comment by cojarbi** [comment]
While in the folder do type: git pull and press enter, this should update

**Comment by Dartellum** [comment]
`git pull` is one way. I did have an instance where I could not compile again after a `git pull`. Removing my git and starting over also did not correct compiling. So, I do this:
```
Make sure on master - git checkout master
git fetch
git reset --hard FETCH_HEAD
git submodule update --init
./make_devel.sh
```

**Reply by 7by24** [reply]
sounds great. thx

**Reply by Mar-85** [reply]
In the main instruction is doesnt mention to use the ./make command again after "git submodule update --init", do i need to do that if just updating?

**Comment by Fernweh-23** [comment]
New to madmax and git.  
How to I run the update?  I think I am doing it all wrong.  I tried to run the commands on both powershell and cmd, both failed.  Is there a specific way to issue the command or am I using the wrong method?

I copy and paste this:
cd {my chia-plotter directory} 
git checkout master
git pull
git submodule update --init
./make_devel.sh
